src/usr/local/www/widgets/javascript/thermal_sensors.js

```diff
@@ -117,37 +120,19 @@ function buildThermalSensorsDataGraph(thermalSensorsData, tsParams, widgetKey) {
 
 }
 
-function updateThermalSensorsDataGraph(thermalSensorsData, tsParams, widgetKey) {
+function updateThermalSensorsDataGraph(thermalSensorsData, widgetKey) {
 	var thermalSensorsArray = new Array();
 
 	if (thermalSensorsData && thermalSensorsData != "") {
 		thermalSensorsArray = thermalSensorsData.split("|");
 	}
 
-	//generate graph for each temperature sensor and append to thermalSensorsHTMLContent string
+	//update thermal sensor values
 	for (var i = 0; i < thermalSensorsArray.length; i++) {
 
 		var sensorDataArray = thermalSensorsArray[i].split(":");
-		var sensorName = sensorDataArray[0].trim();
 		var thermalSensorValue = getThermalSensorValue(sensorDataArray[1]);
 
-
-		//set thresholds
-		if (sensorName.indexOf("cpu") > -1) { //check CPU Threshold config settings
-			warningTemp = tsParams.coreWarningTempThreshold;
-			criticalTemp = tsParams.coreCriticalTempThreshold;
-		} else if (sensorName.indexOf("pch") > -1) { //check PCH Threshold config settings
-			warningTemp = tsParams.pchWarningTempThreshold;
-			criticalTemp = tsParams.pchCriticalTempThreshold;
-		} else { //assuming sensor is for a zone, check Zone Threshold config settings
-			warningTemp = tsParams.zoneWarningTempThreshold;
-			criticalTemp = tsParams.zoneCriticalTempThreshold;
-		}
-
-		if (!tsParams.showFullSensorName) {
-			sensorName = getSensorFriendlyName(sensorName);
-		}
-
```

Comment:
It seems like these should be removed from `buildThermalSensorsDataGraph()` instead.

Reply:
I think it's best to calculate these values when "building", and avoid unnecessary work when "updating". When a user updates the thresholds, this results in a full page reload anyway. As long as the user has only one tab open and doesn't change settings in another tab, the thresholds will always be up to date if we only calculate them once at the beginning.